lms/djangoapps/program_enrollments/docs/decisions/0003-enrollments-into-course-shared-by-programs.rst

```diff
@@ -0,0 +1,60 @@
+Enrollments Into A Course Shared By Multiple Programs
+-----------------------------------------------------
+
+Status
+======
+
+Accepted (circa March 2020)
+
+
+Context
+=======
+
+For the spring term 2020 (Starting Jan 6th), Georgia Tech has created 3
+different Masters programs on edX. One such Masters program is named
+"Other Master Program". This program is linked to all the courses
+offered by both the Cybersecurity and Data Analytics programs.
+This created a situation where every Georgia Tech's course run is
+referenced by two different Masters program at least. We need a
+mechanism to properly control the integrity of student enrollment data
+in light of the cross reference.
+
+The scope of this decision is related to how API would react to enrollment
+requests to a course for a single user, but from more than one programs.
+The decision also covers the scenario where a learner needs to move
+program enrollment from one program to another program.
+
+Decisions
+=========
+
+We update the field "Course Enrollment" of the "ProgramCourseEnrollment"
+model from OneToOne field to a Foreign Key field.
+We also implemented logic to prevent two "active" "programCourseEnrollment"
+records to be written into data store for the same user for the same course
+across any number of affliated programs. This means, only 0 or 1 "active"
+enrollment for a learner and the enrolled course can exist in our data store
+at any given time.
```

Comment:
One of the consequences of this I think we should call out is that we have an unambiguous sense of when we can mark a learner's student_courseenrollment as inactive as a result of a change in the programcourseenrollment. Might make sense to note in the consequences section or as part of the rationale for rejecting "multiple active ProgramCourseEnrollments".

Reply:
👍